Server-side error when archiving content #4291

ashklianko opened the pull request:
Issue is in that when archiving items we also receive update events prior to delete/move events, that triggers fetching of content that is already can't be found in the root

-improved event filtering: we use debounce for all types of events, so when it is time to execute update event handler we check if items in update queue are also can be found in the move/delete queue, and if they do then we remove those items from update queue
-Not executing lazy load handler in dependants list if dialog is closed

ashklianko committed:
Server-side error when archiving content #4291

Issue is in that when archiving items we also receive update events prior to delete/move events, that triggers fetching of content that is already can't be found in the root

-improved event filtering: we use debounce for all types of events, so when it is time to execute update event handler we check if items in update queue are also can be found in the move/delete queue, and if they do then we remove those items from update queue
-Not executing lazy load handler in dependants list if dialog is closed

alansemenov commented on modules/lib/src/main/resources/assets/js/app/event/ServerEventAggregator.ts:
```diff
@@ -47,6 +47,41 @@ export class ServerEventAggregator {
+    private isNotMoveOrDeleteEvent(typeAndRepo: string): boolean {
+        const eventType: string = typeAndRepo.substring(0, typeAndRepo.indexOf(':'));
+        return eventType !== NodeServerChangeType[NodeServerChangeType.MOVE]
+               && eventType !== NodeServerChangeType[NodeServerChangeType.DELETE];
```
Maybe
`return typeAndRepo.startsWith(NodeServerChangeType[NodeServerChangeType.MOVE]) || typeAndRepo.startsWith(NodeServerChangeType[NodeServerChangeType.DELETE]);`?

alansemenov replied:
Or `const eventType: string = typeAndRepo.split(':')[0];`

ashklianko replied:
Updated

ashklianko committed:
Server-side error when archiving content #4291

-small upd

ashklianko committed:
-Issue with an archive routine: content gets updated first and only after that it is moved to /archive root, thus update events come earlier than move events, and if there are lots of content to archive then debounce time between update and move is might be not enough (when move happens we used to filter moved items from the items that are pending updated) and update is fired before filtering occurs
-Fixed by listening Task events: if task is about archiving content then postponing update handler until archive task events stop coming, then move event has a chance to be received and filter moved items from update sequence

ashklianko commented:
Updated, however 'Project node sync failed' keep coming from backend, not related to frontend

alansemenov commented:
Will be fixed in the backend
